kiwi/tasks/system_build.py

```diff
@@ -155,11 +155,12 @@ def process(self):
         system = SystemPrepare(
             self.xml_state, image_root, True
         )
+
         manager = system.setup_repositories()
-        system.install_bootstrap(manager)
-        system.install_system(
-            manager
-        )
+        if self.xml_state.get_bootstrap_packages_sections():
+            system.install_bootstrap(manager)
+        system.install_system(manager)
+
```

Comment:
I agree the condition check is correct, did not look close enough to see that this is checking for all <packages> section marked as bootstrap. I still think the check should be done once in the SystemPrepare class: e.g

```
diff --git a/kiwi/system/prepare.py b/kiwi/system/prepare.py
index fa5a2730..63f99c31 100644
--- a/kiwi/system/prepare.py
+++ b/kiwi/system/prepare.py
@@ -155,6 +155,11 @@ class SystemPrepare(object):
         Install system software using the package manager
         from the host, also known as bootstrapping
         """
+        if not self.xml_state.get_bootstrap_packages_sections():
+            log.warning('No <packages> sections marked as "bootstrap" found')
+            log.info('Processing of bootstrap stage skipped')
+            return
+
         log.info('Installing bootstrap packages')
         bootstrap_packages = self.xml_state.get_bootstrap_packages()
         bootstrap_packages.append(
```